runtime/src/mem/cstdlib/mem-cstdlib.c

```diff
@@ -35,8 +35,11 @@ void chpl_mem_layerInit(void) {
   size_t size;
 
   chpl_comm_desired_shared_heap(&start, &size);
-  if (start || size)
-    chpl_error("set CHPL_MEM to a more appropriate mem type", 0, 0);
+  if (start || size) {
```

Comment:
Teensy nit: probably okay to just check `start`; in other configs we assert that start and size are either both zero or both nonzero.